components/admin_console/custom_plugin_settings/index.ts

```diff
@@ -17,24 +19,26 @@ import {it} from '../admin_definition';
 
 import {appsPluginID} from 'utils/apps';
 
-import CustomPluginSettings from './custom_plugin_settings.jsx';
+import CustomPluginSettings from './custom_plugin_settings';
 import getEnablePluginSetting from './enable_plugin_setting';
 
+type OwnProps = { match: { params: { plugin_id: string } } }
+
 function makeGetPluginSchema() {
     return createSelector(
         'makeGetPluginSchema',
-        (state, pluginId) => state.entities.admin.plugins[pluginId],
-        (state, pluginId) => getAdminConsoleCustomComponents(state, pluginId),
+        (state: GlobalState, pluginId: string) => state.entities.admin.plugins![pluginId],
+        (state: GlobalState, pluginId: string) => getAdminConsoleCustomComponents(state, pluginId),
         (state) => appsFeatureFlagEnabled(state),
-        (plugin, customComponents, appsFeatureFlagIsEnabled) => {
+        (plugin: PluginRedux & {translate?: boolean}, customComponents, appsFeatureFlagIsEnabled) => {
             if (!plugin) {
                 return null;
             }
 
             const escapedPluginId = SchemaAdminSettings.escapePathPart(plugin.id);
             const pluginEnabledConfigKey = 'PluginSettings.PluginStates.' + escapedPluginId + '.Enable';
 
-            let settings = [];
+            let settings: Array<Partial<PluginSetting> & Partial<SchemaAdminSettings>> = [];
```

Comment:
you are already using `PluginSetting` here, so it is probably safe to use it in the `getEnablePluginSetting` function as well (might need to cross-check the usage of it though)

Reply:
As explained above unfortunately `PluginSetting` doesn't include everything.